cdk_eoapi/pgStacInfra.py

```diff
@@ -0,0 +1,167 @@
+from aws_cdk import (
+    Stack,
+    aws_iam,
+    aws_ec2,
+    aws_rds,
+)
+from constructs import Construct
+from cdk_pgstac import (
+    BastionHost,
+    PgStacApiLambda,
+    PgStacDatabase,
+    StacIngestor,
+    TitilerPgstacApiLambda,
+)
+from typing import Union, Optional
+import boto3
+
+
+class pgStacInfraStack(Stack):
+    def __init__(
+        self,
+        scope: Construct,
+        id: str,
+        vpc: aws_ec2.Vpc,
+        stage: str,
+        db_allocated_storage: int,
+        public_db_subnet: bool,
+        db_instance_type: str,
+        stac_api_lambda_name: str,
+        titiler_pgstac_api_lambda_name: str,
+        bastion_host_allow_ip_list: list,
+        bastion_host_create_elastic_ip: bool,
+        titiler_buckets: list,
+        data_access_role_arn: Optional[str],
+        auth_provider_jwks_url: Optional[str],
+        bastion_host_user_data: Union[str, aws_ec2.UserData],
+        **kwargs,
+    ) -> None:
+        super().__init__(scope, id, **kwargs)
+
+        pgstac_db = PgStacDatabase(
+            self,
+            "pgstac-db",
+            vpc=vpc,
+            engine=aws_rds.DatabaseInstanceEngine.postgres(
+                version=aws_rds.PostgresEngineVersion.VER_14
+            ),
+            vpc_subnets=aws_ec2.SubnetSelection(
+                subnet_type=aws_ec2.SubnetType.PUBLIC
+                if public_db_subnet
+                else aws_ec2.SubnetType.PRIVATE_ISOLATED
+            ),
+            allocated_storage=db_allocated_storage,
```

Comment:
@emileten I don't see `allocated_storage` as  prop in the current `cdk-pgstac` `PgStacDatabase`.  Where is this going to?

Reply:
@sharkinsspatial `PgStacDataBaseProps` extend `rds.DataBaseInstanceProps` which itself extends a class that has an `allocatedStorage` parameter https://github.com/aws/aws-cdk/blob/928b64ecbfb68ceaf4e15a5fb434c4b39dc7a4a1/packages/aws-cdk-lib/aws-rds/lib/instance.ts#L926. 

Except if I am making a mistake, in the python version we have to pass `allocated_storage` to use that parameter.